src/Sulu/Bundle/ContentBundle/Resources/config/content_types.xml

```diff
@@ -20,6 +20,7 @@
             <argument type="service" id="sulu_content.smart_content.data_provider.content.query_builder"/>
             <argument type="service" id="logger"/>
             <argument>%sulu.content.type.internal_links.template%</argument>
+            <argument type="expression">parameter('sulu.context') === 'website' and (!container.hasParameter('sulu.preview') or !parameter('sulu.preview'))</argument>
```

Comment:
One thing that comes to mind is to create a new parameter somewhere, which is the combination of these two. This way I wouldn't have to repeat myself for the smart content.